contracts/truefi/TrueFarm.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {TrueDistributor} from "./TrueDistributor.sol";
+
+contract TrueFarm {
+    using SafeMath for uint256;
+
+    ERC20 public stakingToken;
+    ERC20 public rewardToken;
+    TrueDistributor public trueDistributor;
+
+    uint256 public totalStaked;
+    mapping(address => uint256) public staked;
+
+    uint256 public cumulativeRewardPerToken;
+    mapping(address => uint256) public previousCumulatedRewardPerToken;
+    mapping(address => uint256) public claimableReward;
+
+    uint256 public totalClaimedRewards;
+    uint256 public totalFarmRewards;
+
+    constructor(ERC20 _stakingToken, TrueDistributor _trueDistributor) public {
+        stakingToken = _stakingToken;
+        trueDistributor = _trueDistributor;
+        rewardToken = _trueDistributor.token();
+    }
+
+    function stake(uint256 amount) public update {
+        require(stakingToken.transferFrom(msg.sender, address(this), amount));
+        staked[msg.sender] += amount;
+        totalStaked += amount;
+    }
+
+    function unstake(uint256 amount) public update {
+        require(amount <= staked[msg.sender], "insufficient staking balance");
+        require(stakingToken.transfer(msg.sender, amount));
+        staked[msg.sender] -= amount;
+        totalStaked -= amount;
+    }
+
+    function claim() public update {
+        rewardToken.transfer(msg.sender, claimableReward[msg.sender]);
+        totalClaimedRewards = totalClaimedRewards.add(claimableReward[msg.sender]);
+        claimableReward[msg.sender] = 0;
+    }
+
+    modifier update() {
+        trueDistributor.distribute(address(this));
+        uint256 newTotalFarmRewards = rewardToken.balanceOf(address(this)).add(totalClaimedRewards).mul(1e30);
+        uint256 totalBlockReward = newTotalFarmRewards.sub(totalFarmRewards);
+        totalFarmRewards = newTotalFarmRewards;
+        if (totalStaked > 0) {
+            cumulativeRewardPerToken += totalBlockReward.div(totalStaked);
+        }
```

Comment:
I think there's a problem with the first staker to come when some rewards have already accrued. Let's assume:
* distributor has awarded the farm with 1000 tokens, but no stakers yet
* first staker comes, his rewards are 0
* when he claims, he gets only the rewards since he staked, the first 1000 tokens are lost

Reply:
I think it's not a bug, but a feature 😁